Replace --delete with --delete-destination-extra in transfer commands (#1037)
globus/globus-cli: changelog.d/20241015_111820_max.tuecke_sc_35029_deprecate_delete_flag.md

> Added
> ~~~~~~~

Reviewer: In restructured text it's expected the markup for the header is the same number of characters
```suggestion
Added
~~~~~
```

> - Added new `--delete-destination-extra` flag to `globus timer create transfer` and `globus transfer` that mirrors the existing `--delete` flags behavior. (:pr:`1037`)

Reviewer 2: ```suggestion
- Added new `--delete-destination-extra` flag to `globus timer create transfer` and `globus transfer` that mirrors the existing `--delete` flags' behaviors. (:pr:`1037`)
```

> - Added new `--delete-destination-extra` flag to `globus timer create transfer` and `globus transfer` that mirrors the existing `--delete` flags behaviors. (:pr:`1037`)

Reviewer: Double backticks are used in restructured text to denote source code. Stephen's [changelog fragment](https://raw.githubusercontent.com/globus/globus-sdk-python/refs/heads/main/changelog.d/20240917_121333_sirosen_add_token_response_classes.rst) is an example.
```suggestion
- Added new ``--delete-destination-extra`` flag to ``globus timer create transfer`` and ``globus transfer`` that mirrors the existing ``--delete`` flags behaviors. (:pr:`1037`)
```

> - Added new `--delete-destination-extra` flag to `globus timer create transfer` and `globus transfer` that mirrors the existing `--delete` flags behaviors. (:pr:`1037`)

Reviewer 2: In addition to Mike's comments, this is missing an apostrophe.

```suggestion
- Added new `--delete-destination-extra` flag to `globus timer create transfer` and `globus transfer` that mirrors the existing `--delete` flags' behaviors. (:pr:`1037`)
```

> - Added deprecation warning to the old `--delete` flag. (:pr:`1037`)

Reviewer: ```suggestion
- Added deprecation warning to the old ``--delete`` flag. (:pr:`1037`)
```
